### PR TITLE
Update openshift api

### PR DESCRIPTION
**Pull request by kevinrizza**
This pull requests migrates the vendored dependency of the `openshift/api` project to the latest.
This is being done in preparation for the release of Openshift 4.1

-Update the vendored dependency to github.com/openshift/api
`dep ensure -update github.com/openshift/api`
-Update the vendored dependency to github.com/openshift/client-go as a
prerequisite
`dep ensure -update github.com/openshift/client-go`
-Updated all references of (`"github.com/openshift/api/config/v1"`)
configv1.OperatorFailing to configv1.OperatorDegraded
-Updated comments, functions, and variable names to match

### CODE REVIEW

**Comment by kevinrizza** [comment]
/cc @soltysh @jwforres @awgreene @galletti94 @anik120 @SamiSousa @jeremylinlin 

Trying to put as many eyes on this one as possible.

Thanks guys.

File: pkg/status/status.go [hunk 1]
```diff
@@ -339,7 +339,7 @@ func (s *status) monitorClusterStatus() {
 		case <-s.stopCh:
 			// If the stopCh is closed, the operator will exit and CO should
 			// be set to failing.
```
Comment (awgreene) [inline pkg/status/status.go]: This should say
`// be set to degrading`

Comment (kevinrizza) [reply pkg/status/status.go]: @awgreene Nice catch! Updated in latest.

**Pull request merged**